universe/interface.go

```diff
@@ -41,14 +41,19 @@ type Identifier struct {
 	GroupKey *btcec.PublicKey
 }
 
-// String returns a string representation of the ID.
-func (i *Identifier) String() string {
+// Bytes returns a bytes representation of the ID.
+func (i *Identifier) Bytes() [32]byte {
 	if i.GroupKey != nil {
-		h := sha256.Sum256(schnorr.SerializePubKey(i.GroupKey))
-		return hex.EncodeToString(h[:])
+		return sha256.Sum256(schnorr.SerializePubKey(i.GroupKey))
 	}
 
-	return hex.EncodeToString(i.AssetID[:])
+	return i.AssetID
+}
+
+// String returns a string representation of the ID.
+func (i *Identifier) String() string {
+	idBytes := i.Bytes()
+	return hex.EncodeToString(idBytes[:])
```

Comment:
nit: could use `fn.ByteSlice(i.Bytes())` directly.

Reply:
Do you want me to write the function like this?
```
// String returns a string representation of the ID.
func (i *Identifier) String() string {
	idBytes := fn.ByteSlice(i.Bytes())
	return hex.EncodeToString(idBytes)
}
```

Or do you mean like this?
```
// String returns a string representation of the ID.
func (i *Identifier) String() string {
	return hex.EncodeToString(fn.ByteSlice(i.Bytes()))
}
```